drivers/power_gating/power_gating_intel.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <drivers/gpio.h>
+#include <pm/device.h>
+#include <pm/device_runtime.h>
+#include <pm/power_gating.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(power_mgmt_ACE, LOG_LEVEL_INF);
+
+static int poll_for_register_delay(uint32_t reg, uint32_t mask, uint32_t val, uint64_t us)
+{
+	if (!WAIT_FOR((sys_read32(reg) & mask) != val, us, k_busy_wait(1))) {
+		LOG_ERR("%s poll timeout reg %u mask %u val %u us %u", __func__, reg, mask, val,
+			(uint32_t)us);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int device_set_power_enable(struct pm_pg_registers reg, bool power_enable, uint64_t us)
+{
+	int ret;
+	uint32_t spa_mask = BIT(reg.SPA_bit);
+	uint32_t cpa_mask = BIT(reg.CPA_bit);
+
+	if (power_enable)
+		sys_write32(sys_read32(reg.wr_address) | spa_mask, reg.wr_address);
+	else
+		sys_write32(sys_read32(reg.wr_address) & ~(spa_mask), reg.wr_address);
+
+	/* Check if powered on. */
+	ret = poll_for_register_delay(reg.rd_address, cpa_mask, spa_mask, us);
+
+	if (ret)
+		LOG_WRN("%s warning: timeout", __func__);
+
+	return ret;
+}
+
+#ifdef CONFIG_PM_DEVICE_POWER_DOMAIN
+
+int set_pd_power_enable(struct pm_pg_registers *reg, bool power_enable)
```

Comment:
```suggestion
int set_pd_power_enable(struct pm_pg_registers *reg, bool power_enable)
```

Ditto, actually here is the one responsible for the power domain. The namespace for this function is not right.

Actually it seems that this is an internal function, just reduce (`static`) its scope then.

Reply:
agree, changed